modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/management_bar/components/NavBar.js

```diff
@@ -22,12 +22,14 @@ import FiltersDropdown from './FiltersDropdown';
 import MainSearch from './MainSearch';
 
 function NavBar({creationMenu, showSearch, views}) {
-	const {state} = useAppState();
+	const {
+		state: {filters},
+	} = useAppState();
 
 	return (
 		<nav className="management-bar management-bar-light navbar navbar-expand-md">
 			<div className="container-fluid container-fluid-max-xl">
-				{state.filters.length && (
+				{!!filters.length && (
```

Comment:
I know you discussed this in the other PR. I stagger when reading this myself, but I agree we better start getting used to it! 🤓

Reply:
Maybe we need a lint for this, because it is too easy to get it wrong. I mean, it's an impedance mismatch between JS (where `0` is falsy) and React's semantics (where `0` is rendered as-is). So, you do something like `maybeZero && blah`, with the entirely reasonable expectation that it render nothing or `blah`, and instead you find it renders `0` or `blah`.

Thinking of something like flagging `whatever.length &&` inside JSX (specifically inside a `JSXExpressionContainer`). I actually don't care where we autofix to `whatever.length ? blah : null` or to `!!whatever.length && blah`, just that we have something in place to avoid at least some of the errors that can sneak through quite easily.

I made [this issue](https://github.com/liferay/eslint-config-liferay/issues/194) so we can discuss options.